updater/lib/dependabot/service.rb

```diff
@@ -39,6 +42,25 @@ def record_update_job_error(error_type:, error_details:, dependency: nil)
       client.record_update_job_error(error_type: error_type, error_details: error_details)
     end
 
+    # This method wraps the Raven client as the Application error tracker
+    # the service uses to notice errors.
+    #
+    # This should be called as an alternative/in addition to record_update_job_error
+    # for cases where an error could indicate a problem with the service.
+    def capture_exception(error:, job: nil, dependency: nil, tags: {}, extra: {})
+      Raven.capture_exception(
+        error,
+        {
+          tags: tags,
+          extra: extra.merge({
+            update_job_id: job&.id,
+            package_manager: job&.package_manager,
+            dependency_name: dependency&.name
+          }.compact)
```

Comment:
🔎 I understand the impulse to send only those key/value pairs for which we have data, but I wonder if the absence of values is valuable for understanding context.

As a 🤏 cheap argument, I'd be able to glean much more insight from:

```ruby
{
  update_job_id: 1,
  dependency_name: "dependabot-fortran",
  package_manager: nil
}
```

than
```ruby
{
  update_job_id: 1,
  dependency_name: "dependabot-fortran"
}
```

Reply:
That's fair, I think we should actually expect `dependency_name` to be passed/present move often -but- I'm err'ing on the side of maintaining existing behaviour for now so I'd prefer to pass on this to avoid test churn for now.